packages/driver/cypress/integration/e2e/multi_domain_spec.ts

```diff
@@ -312,6 +314,42 @@ describe('multi-domain', { experimentalSessionSupport: true, experimentalMultiDo
       cy.switchToDomain('foobar.com', () => {}, () => {})
     })
 
+    it('waits for all logs to finish streaming in from switchToDomain, expecting commands to not be pending', (done) => {
+      const domain = 'foobar.com'
+      const logsAddedNeedingUpdate = {}
+      const logsChangedGivingUpdate = {}
+
+      cy.on('log:added', (addedLog) => {
+        if (!addedLog.ended && addedLog?.id.includes(domain)) {
+          logsAddedNeedingUpdate[addedLog.id] = addedLog
+        }
+      })
+
+      cy.on('log:changed', (changedLog) => {
+        if (changedLog.ended && changedLog?.id.includes(domain)) {
```

Comment:
If it's possible for `changedLog` to be undefined, shouldn't both references here use the optional chaining operator?

Reply:
Technically, I think so. We should put the optional chainer here to be safe. Will update!

Reply:
I'm also going to add this to `addedLog`